File: src/test/resources/oracle/commits/rm-action-test-0a5085070f80f44c03645e1557c2bba8da3b335e/typescript/CustomerProfile.ts
export class CustomerProfile {
    private fullName: string;
    private street: string;
    private city: string;
    private postalCode: string;
    public loyaltyPoints: number;

    constructor(fullName: string, street: string, city: string, postalCode: string, loyaltyPoints: number) {
        this.fullName = fullName;
        this.street = street;
        this.city = city;
        this.postalCode = postalCode;
        this.loyaltyPoints = loyaltyPoints;
    }

    mailingLabel(): string {
        return `${this.fullName}\n${this.street}\n${this.city} ${this.postalCode}`;
    }

    canReceivePromotion(optedIn: boolean, bouncedEmail: boolean, purchaseCount: number): boolean {
        if (optedIn && !bouncedEmail && purchaseCount > 0) {
            return true;
        }
        return false;
    }

    addPurchase(amountCents: number): void {
        const points = Math.floor(amountCents / 100);
        this.loyaltyPoints += points;
    }
}

File: src/test/resources/oracle/commits/rm-action-test-1600b87d81dac9889f5603e07849e6bc89995332/kotlin/CustomerProfile.kt
class CustomerProfile(
    private val fullName: String,
    private val street: String,
    private val city: String,
    private val postalCode: String,
    var loyaltyPoints: Int
) {
    fun mailingLabel(): String {
        return "$fullName\n$street\n$city $postalCode"
    }

    fun canReceivePromotion(optedIn: Boolean, bouncedEmail: Boolean, purchaseCount: Int): Boolean {
        if (optedIn && !bouncedEmail && purchaseCount > 0) {
            return true
        }
        return false
    }

    fun addPurchase(amountCents: Int) {
        val points = amountCents / 100
        loyaltyPoints += points
    }
}
